cmake/flash/CMakeLists.txt

```diff
@@ -13,6 +13,52 @@ function(runner_yml_write content)
     )
 endfunction()
 
+function(get_runners_prop prop out_var)
+  # Get property 'prop' from runners_yaml_props_target. An optional
+  # third argument can be used to provide a default value if the value
+  # is not found (set to ...-NOTFOUND).
+
+  get_target_property(out runners_yaml_props_target "${prop}")
+
+  if(("${ARGC}" EQUAL 3) AND ("${out}" STREQUAL "out-NOTFOUND"))
+    set("${out_var}" "${ARGV2}" PARENT_SCOPE)
+  else()
+    set("${out_var}" "${out}" PARENT_SCOPE)
+  endif()
+endfunction()
+
+function(runners_yaml_append_config)
+  # Append the common configuration values to the relevant property target.
+
+  runners_yaml_append("\n# Common runner configuration values.")
+  runners_yaml_append("config:")
+  runners_yaml_append("  board_dir: ${BOARD_DIR}")
+  get_runners_prop(elf_file elf "${KERNEL_ELF_NAME}")
+  runners_yaml_append("  # Build outputs:")
+  runners_yaml_append("  elf_file: ${elf}")
+  if(CONFIG_BUILD_OUTPUT_HEX)
+    get_runners_prop(hex_file hex "${KERNEL_HEX_NAME}")
+    runners_yaml_append("  hex_file: ${hex}")
+  endif()
+  if(CONFIG_BUILD_OUTPUT_BIN)
+    get_runners_prop(bin_file bin "${KERNEL_BIN_NAME}")
+    runners_yaml_append("  bin_file: ${bin}")
+  endif()
+
+  if (DEFINED CMAKE_GDB OR DEFINED OPENOCD OR DEFINED OPENOCD_DEFAULT_PATH)
+    runners_yaml_append("  # Host tools:")
+  endif()
+  if (DEFINED CMAKE_GDB)
+    runners_yaml_append("  gdb: ${CMAKE_GDB}")
+  endif()
+  if (DEFINED OPENOCD)
+    runners_yaml_append("  openocd: ${OPENOCD}")
+  endif()
+  if(DEFINED OPENOCD_DEFAULT_PATH)
```

Comment:
@tejlmand I want to check this section with you.

Is, for example `if(DEFINED OPENOCD)` OK to use here, or should I be using `if(OPENOCD)`? Similarly for other host tools.

Might these get defined to `OPENOCD-NOTFOUND` or so in some cases if they are not found? If so, I think `DEFINED` should maybe be dropped.

Reply:
This is one of the tricky parts in CMake, cause if it is just a variable that has NEVER been defined, then using `if(DEFINED never_defined_var)` and `if(never_defined_var)` will both return false.

But if `find_program(OPENOCD openocd)` has been called earlier, then `OPENOCD` will actually be defined, but having the value `OPENOCD-NOTFOUND`, and in that case:
`if(OPENOCD)` will return false (because `OPENOCD-NOTFOUND` is a false value)
but 
`if(DEFINED OPENOCD)` will return True, because it is  defined with a value.

But if `find_program(OPENOCD ...)` was never executed, both return false.

So correct checking in this case is:
```
if (OPENOCD)
    runners_yaml_append("  openocd: ${OPENOCD}")
endif()
```

Reply:
Great, fixed, thanks.